src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,180 @@
+
+julia> idx = Oscar._quantum_automorphism_group_indices(M,:bases);
+
+julia> length(idx)
+1920
+```
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  matroid_groundset(M) isa Vector{Int} ? nothing : M=Matroid(Oscar.pm_object(M))  
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  b    = [Vector{Int}[] for _ in 1:n]
+  nb   = [Vector{Int}[] for _ in 1:n]
+  rels = [Vector{Tuple{Int,Int}}[] for _ in 1:n]
+
+  sets  = getproperty(Oscar, structure)(M)
+  sizes = unique!(map(length, sets))
+  tempGrdSet = reduce(vcat,[grdSet for i in 1:n])
+
+  for size in sizes 
+    size == 0 && continue
+    powerSet = unique(sort.(subsets(tempGrdSet, size)))
+
+    setsOfSize = filter(x->length(x)==size, sets)
+    nonSets = setdiff(powerSet, setsOfSize) 
+
+    S = symmetric_group(size)
+
+    for set in setsOfSize
+      append!(b[size], [[set[g(i)] for i in 1:size] for g in S])
+    end
+    for nonset in nonSets
+      append!(nb[size], [[nonset[g(i)] for i in 1:size] for g in S])
+    end
+    for set in b[size]
+      for nonset in nb[size]
+        push!(rels[size], [(set[i],nonset[i]) for i in 1:size])
+        push!(rels[size], [(nonset[i],set[i]) for i in 1:size])
+      end
+    end
+  end
+  rels = unique!.(rels)
+  return Vector{Vector{Tuple{Int,Int}}}(reduce(vcat,rels))
+end
+
+@doc raw"""
+    quantum_automorphism_group(M::Matroid, structure::Symbol=:bases)
+
+Return the Ideal that defines the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> G = complete_graph(4)
+Undirected graph with 3 nodes and the following edges:
+(2, 1)(3, 1)(3, 2)
+
+julia> M = cycle_matroid(G);
+
+julia> qAut = quantum_automorphism_group(M,:bases);
+
+julia> length(gens(qAut))
+23448
+```
+"""
+function quantum_automorphism_group(
+  M::Matroid,
+  structure::Symbol=:bases)
+
+  relation_indices = _quantum_automorphism_group_indices(M, structure)
+  n = length(M)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n, n)),[2, 1])
+
+  new_relations = elem_type(A)[prod(gen -> u[gen[1], gen[2]], relation; init=one(A)) for relation in relation_indices]
+  return ideal([gens(SN)..., new_relations...])
+
+end
+
+@doc raw"""
+    quantum_automorphism_group(G::Graph)
+
+Return the Ideal that defines the quantum automorphism group of a graph.
+
+# Examples
+```jldoctest
+julia> G = graph_from_edges([[1, 2], [2, 4]]);
+
+julia> qAut = quantum_automorphism_group(G);
+
+julia> length(gens(qAut))
+168
+```
+"""
+function quantum_automorphism_group(G::Graph{Undirected})
+
+  n = nv(G)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n, n)),[2, 1])
+
+  nonedges = Tuple{Int,Int}[(i, j) for i in 1:n for j in i:n if !has_edge(G,i,j)] # This should be n^2 - |E| many nonedges
```

Comment:
What is `E`? If you can make |E| concrete then it might be better to turn this comment into an assertion?

Here's a guess
```suggestion
  nonedges = Tuple{Int,Int}[(i, j) for i in 1:n for j in i:n if !has_edge(G,i,j)]
  @assert length(nonedges) == n^2 - n_edges(G)
```

Reply:
(To be clear: the code is overall in pretty great shape now, feel free to ignore my additional suggestions)